Add missing Auth module unit tests

bobanm opened the pull request:
### What was the problem?

This PR resolves #7826

### How was it solved?

Added all unit tests from the related issue, except for:

> Account registered with 4 optional keys. Number of signatures required 2. Transaction successful when optional signatures ordered and present. Test with different positions of keys.

Instead of creating a new 2-of-4 with 0 mandatory signer test case, I updated the existing 1-of-2 with 0 mandatory signer test case, to check for different positions of keys.

Bonus: added a test suite for `auth/utils` file with a test case for `verifySignature()` 😎

### How was it tested?

All tests pass 👌🏻

shuse2 commented on framework/test/unit/modules/auth/register_multisignature.spec.ts:
```diff
@@ -72,6 +72,78 @@ describe('Register Multisignature command', () => {
+			expect(result.status).toBe(VerifyStatus.OK);
+		});
+
+		it('should return status OK when the total number of keys between mandatory and optional is 64', async () => {
```
```suggestion
		it('should return status OK when the total number of mandatory and optional keys is 64', async () => {
```

shuse2 reviewed (approved):
global variable mutation is introduced in this PR as well, but if it is resolved in the previous PR, it should automatically be fixed in this PR as well

bobanm commented:
> global variable mutation is introduced in this PR as well, but if it is resolved in the previous PR, it should automatically be fixed in this PR as well

@shuse2 I have removed global variable mutation in the parent PR and rebased this branch on the updated parent branch.
